p2p/src/main/java/bisq/network/p2p/storage/P2PDataStorage.java

```diff
@@ -380,50 +390,50 @@ public boolean addProtectedStorageEntry(ProtectedStorageEntry protectedStorageEn
             return false;
         }
 
-        boolean sequenceNrValid = isSequenceNrValid(protectedStorageEntry.getSequenceNumber(), hashOfPayload);
-        boolean result = sequenceNrValid &&
-                checkPublicKeys(protectedStorageEntry, true)
-                && checkSignature(protectedStorageEntry);
+        // TODO: Combine with hasSequenceNrIncreased check, but keep existing behavior for now
+        if(!isSequenceNrValid(protectedStorageEntry.getSequenceNumber(), hashOfPayload))
+            return false;
+
+        // Verify the ProtectedStorageEntry is well formed and valid for the add operation
+        if (!checkPublicKeys(protectedStorageEntry, true) || !checkSignature(protectedStorageEntry))
+            return false;
 
         boolean containsKey = map.containsKey(hashOfPayload);
-        if (containsKey) {
-            result = result && checkIfStoredDataPubKeyMatchesNewDataPubKey(protectedStorageEntry.getOwnerPubKey(), hashOfPayload);
-        }
 
-        // printData("before add");
-        if (result) {
-            boolean hasSequenceNrIncreased = hasSequenceNrIncreased(protectedStorageEntry.getSequenceNumber(), hashOfPayload);
+        // If we have already seen an Entry with the same hash, verify the new Entry has the same owner
+        if (containsKey && !checkIfStoredDataPubKeyMatchesNewDataPubKey(protectedStorageEntry.getOwnerPubKey(), hashOfPayload))
+            return false;
 
-            if (!containsKey || hasSequenceNrIncreased) {
-                // At startup we don't have the item so we store it. At updates of the seq nr we store as well.
-                map.put(hashOfPayload, protectedStorageEntry);
-                hashMapChangedListeners.forEach(e -> e.onAdded(protectedStorageEntry));
-                // printData("after add");
-            } else {
-                log.trace("We got that version of the data already, so we don't store it.");
-            }
+        boolean hasSequenceNrIncreased = hasSequenceNrIncreased(protectedStorageEntry.getSequenceNumber(), hashOfPayload);
 
-            if (hasSequenceNrIncreased) {
-                sequenceNumberMap.put(hashOfPayload, new MapValue(protectedStorageEntry.getSequenceNumber(), System.currentTimeMillis()));
-                // We set the delay higher as we might receive a batch of items
-                sequenceNumberMapStorage.queueUpForSave(SequenceNumberMap.clone(sequenceNumberMap), 2000);
+        // If we have seen a more recent operation for this payload, we ignore the current one
```

Comment:
```suggestion
        // If we have seen a more recent version of this payload, we ignore the current one
```

?

Reply:
We don't necessarily see a different version of the payload as the contents are identical, but we do see different operations on that payload that increment the sequence number.

The producers of change (add/remove/refresh) always increment the SeqNr so it acts like a timestamp for which operations take precedence over one another. For example, a Remove operation with SeqNr 2 is superseded by an add with SeqNr 3.

In that way of thinking, it made sense to me that the SeqNrs are more tied to the operations that occur and less about the payload version.

Hope that helps.